### PR TITLE
Use pattern-matching to link to typed api docs

### PR DESCRIPTION
**Pull request by raboof**
Depends on https://github.com/lightbend/sbt-paradox-apidoc/pull/15

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/actor-lifecycle.md
Comment (chbatey) [inline akka-docs/src/main/paradox/typed/actor-lifecycle.md]: this works? nice!

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-docs/src/main/paradox/typed/actor-lifecycle.md
Comment (raboof) [reply akka-docs/src/main/paradox/typed/actor-lifecycle.md]: once lightbend/sbt-paradox-apidoc#15 is merged it will :smile:

File: akka-docs/src/main/paradox/typed/distributed-data.md
Comment (patriknw) [inline akka-docs/src/main/paradox/typed/distributed-data.md]: the main documentation for the `Replicator` is on the `class` not the companion, what will be used in the link?

Comment (raboof) [reply akka-docs/src/main/paradox/typed/distributed-data.md]: In typed, there is only the object, there is no class (so before this link caused a 404 for the scaladsl).

Comment (patriknw) [reply akka-docs/src/main/paradox/typed/distributed-data.md]: ah, I looked at untyped

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by raboof** [comment]
(thanks for the reviews, but needs https://github.com/lightbend/sbt-paradox-apidoc/pull/15 to be reviewed/merged first before this can go in ;) )

**Comment by patriknw** [comment]
PLS BUILD

**Comment by patriknw** [comment]
can be merged after additional validation build

**Comment by raboof** [comment]
changes already applied on master

**Pull request closed without merging**
